satviz-producer/src/main/java/edu/kit/satviz/producer/ProducerApplication.java

```diff
@@ -1,4 +1,78 @@
 package edu.kit.satviz.producer;
 
+import edu.kit.satviz.common.Constraint;
+import edu.kit.satviz.common.ConstraintValidationException;
+import edu.kit.satviz.network.ProducerConnection;
+import edu.kit.satviz.producer.cli.ProducerCli;
+import edu.kit.satviz.producer.cli.ProducerConstraints;
+import edu.kit.satviz.producer.cli.ProducerParameters;
+import edu.kit.satviz.producer.mode.ProofMode;
+import edu.kit.satviz.producer.mode.SolverMode;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+
+/**
+ * The main class of the producer application.
+ */
 public class ProducerApplication {
+
+  private static final Logger logger = Logger.getLogger("Producer");
+
+  // new modes can be added to this list
+  private static final List<ProducerMode> SUPPORTED_MODES
+      = List.of(new ProofMode(), new SolverMode());
+
+  /**
+   * The entry point of the application.
+   *
+   * @param args The command line args, as defined by the CLI specification.
+   */
+  public static void main(String[] args) {
+    ProducerParameters parameters = parseArgs(args);
+    validateArgs(parameters);
+    ProducerMode selectedMode = SUPPORTED_MODES.stream()
+        .filter(mode -> mode.isSet(parameters))
+        .findFirst()
+        .orElseThrow();
+
+    try {
+      logger.info("Opening clause source");
+      ProducerModeData data = selectedMode.apply(parameters);
+      logger.info("Clause source opened");
+      ProducerConnection connection = new ProducerConnection(
+          parameters.getHost(), parameters.getPort());
+      connection.register(new SourceControlConnectionListener(connection, data.source()));
+      logger.info("Waiting for network connection...");
+      connection.establish(data.id());
+    } catch (SourceException e) {
```

Comment:
Hier sollte die Netzwerk-Verbindung noch korrekt geschlossen werden
```suggestion
    } catch (SourceException e) {
        connection.terminateFailed(e.getMessage());
```

Reply:
Die Exception kommt hier nur von `selectedMode.apply(parameters)`. Die Verbindung ist zu dem Zeitpunkt noch nicht erstellt

Reply:
Ah richtig, das was mir gefehlt hatte ist in SourceControlConnectionListener Zeile 49

Reply:
Hab jetzt aber noch gesehen, dass an der Stelle in dem einen Exception case die connection nicht beendet wurde, hab das noch hinzugefügt.